### PR TITLE
[4737] Editing allowed when item is locked via PageBuilder 

### PR DESCRIPTION
**Pull request by aachavarria**
https://github.com/craftercms/craftercms/issues/4737

### CODE REVIEW

File: ui/app/src/utils/content.ts [hunk 1]
```diff
@@ -124,6 +124,24 @@ export function isImage(path: string): boolean {
...
+export function getComputedEditMode({
+  item,
+  username,
+  editMode
+}: {
+  item: DetailedItem;
+  username: string;
+  editMode: boolean;
+}): boolean {
+  return isItemLockedForMe(item, username) === true || hasEditAction(item.availableActions) === false
```
Comment (rart) [inline ui/app/src/utils/content.ts:140]: Item could be null
```suggestion
  return isItemLockedForMe(item, username) === true || hasEditAction(item?.availableActions) === false
```

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 2]
```diff
@@ -315,19 +316,22 @@ export function PreviewConcierge(props: any) {
...
-  }, [dispatch, currentItemPath, site]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [items[currentItemPath], editMode, user.username]);
 
   useEffect(() => {
-    if (write === false && editMode) {
-      getHostToGuestBus().next({ type: HOST_CHECK_IN, payload: { editMode: false } });
+    if (currentItemPath && site) {
+      dispatch(completeDetailedItem({ path: currentItemPath }));
     }
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx]: Could we omit re-loading the detailed item?

File: ui/app/src/modules/Preview/PreviewConcierge.tsx [hunk 3]
```diff
@@ -438,7 +442,10 @@ export function PreviewConcierge(props: any) {
               dispatch(setPreviewChoice({ site, choice: '2' }));
             }
 
-            getHostToGuestBus().next({ type: HOST_CHECK_IN, payload: { editMode, highlightMode } });
+            getHostToGuestBus().next({
+              type: HOST_CHECK_IN,
+              payload: { editMode: false, highlightMode }
```
Comment (rart) [inline ui/app/src/modules/Preview/PreviewConcierge.tsx:439]: How come editMode is hard set to false here?

**Review by rart — CHANGES_REQUESTED** [review]

File: ui/app/src/utils/content.ts (hunk 1 above)
Comment (aachavarria) [reply ui/app/src/utils/content.ts:140]: this function should no receive a null item

Comment (aachavarria) [reply ui/app/src/utils/content.ts:140]: it is not in the interface, the component using it should validate the item

File: ui/app/src/modules/Preview/PreviewConcierge.tsx (hunk 3 above)
Comment (aachavarria) [reply ui/app/src/modules/Preview/PreviewConcierge.tsx:439]: we talked about this yesterday, guest is going to start false and the effect on the top of previewConcierge is calculating the getComputedEditMode, and updating Guest

**Review by rart — APPROVED** [review]

**Pull request merged**